src/codeu/chat/server/Controller.java

```diff
@@ -22,20 +22,37 @@
 import codeu.chat.common.RandomUuidGenerator;
 import codeu.chat.common.RawController;
 import codeu.chat.common.User;
+import codeu.chat.database.Database;
 import codeu.chat.util.Logger;
 import codeu.chat.util.Time;
 import codeu.chat.util.Uuid;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBCursor;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
 
 public final class Controller implements RawController, BasicController {
 
   private static final Logger.Log LOG = Logger.newLog(Controller.class);
 
   private final Model model;
+  private final Database database;
+
   private final Uuid.Generator uuidGenerator;
+  private final Uuid serverId;
 
-  public Controller(Uuid serverId, Model model) {
+  public Controller(Uuid serverId, Model model, Database database) {
     this.model = model;
     this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+    this.serverId = serverId;
+    this.database = database;
+
+    loadDatabase();
```

Comment:
A common piece of wisdom is:  don't do any "real work" in a constructor. If, instead, you keep constructors private and use a static factory method as the way that other collaborators make a new instance, it actually works out better: that factory method can do all kinds of work until the things the instance will need are ready, then pass those things to the constructor which only has to initialize fields.

One of the main reasons for this is that constructors are weird. When you write an instance method you expect to be acting on a fully-formed, correct, valid instance of your type. But for any instance method that's being called from the constructor, you have to be careful! That might not be a fully-initialized instance. In the static setup described above you keep the time to an absolute minimum when there is a weird partially-constructed instance in existence.